tests/blackbox_asserts.py

```diff
@@ -0,0 +1,298 @@
+    """Instantiate and return a KMD client object."""
+    return kmd.KMDClient(kmd_token, kmd_address)
+
+
+def _indexer_client(
+    indexer_address="http://localhost:8980", indexer_token="a" * 64
+) -> indexer.IndexerClient:
+    """Instantiate and return Indexer client object."""
+    return indexer.IndexerClient(indexer_token, indexer_address)
+
+
+def algod_with_assertion():
+    algod = _algod_client()
+    assert algod.status(), "algod.status() did not produce any results"
+    return algod
+
+
+def mode_to_execution_mode(mode: Mode) -> blackbox.ExecutionMode:
+    if mode == Mode.Application:
+        return blackbox.ExecutionMode.Application
+    if mode == Mode.Signature:
+        return blackbox.ExecutionMode.Signature
+
+    raise Exception(f"Unknown mode {mode} of type {type(mode)}")
+
+
+def blackbox_pyteal(subr: SubroutineFnWrapper, mode: Mode) -> Callable[..., Expr]:
+    """Functor producing ready-to-compile PyTeal programs from annotated subroutines
+
+    Args:
+        subr: annotated subroutine to wrap inside program.
+            Note: the `input_types` parameters should be supplied to @Subroutine() annotation
+        mode: type of program to produce: logic sig (Mode.Signature) or app (Mode.Application)
+
+    Returns:
+        a function that called with no parameters -e.g. result()-
+        returns a PyTeal expression compiling to a ready-to-test TEAL program.
+
+    The return type is callable in order to adhere to the API of end-to-end unit tests.
+
+    Generated TEAL code depends on the mode, subroutine input types, and subroutine output types.
+    * logic sigs:
+        * input received via `arg i`
+        * args are converted (cf. "input conversion" below) and passed to the subroutine
+        * subroutine output is not logged (log is not available)
+        * subroutine output is converted (cf "output conversion" below)
+    * apps:
+        * input received via `txna ApplicationArgs i`
+        * args are converted (cf. "input conversion" below) and passed to the subroutine
+        * subroutine output is logged after possible conversion (cf. "logging conversion")
+        * subroutine output is converted (cf "output conversion" below)
+    * input conversion:
+        * Empty input array:
+            do not read any args and call subroutine immediately
+        * arg of TealType.bytes and TealType.any:
+            read arg and pass to subroutine as is
+        * arg of TealType.uint64:
+            convert arg to int using Btoi() when received
+    * output conversion:
+        * TealType.uint64:
+            provide subroutine's result to the top of the stack when exiting program
+        * TealType.bytes:
+            convert subroutine's result to the top of the stack to its length and then exit
+        * TealType.none or TealType.anytype:
+            push Int(1337) to the stack as it is either impossible (TealType.none),
+            or unknown at compile time (TealType.any) to convert to an Int
```

Comment:
What is the point of doing this?

Reply:
I want to have a way for the user to check that TEAL that was created for a subroutine is functioning as expected, even if the subroutine doesn't return anything (`TealType.none`), or the its type is something I can't actually return automatically (`TealType.anytype`). So I'm providing the sentinel 1337 as a signal that the PyTEAL generator works fine, and we -in fact- got no output. It's almost impossible to prove that something _doesn't exist_ but this tries as hard to do that as it can.